underfs/cos/src/main/java/alluxio/underfs/cos/COSOutputStream.java

```diff
@@ -0,0 +1,164 @@
+    Preconditions.checkArgument(bucketName != null && !bucketName.isEmpty(),
+        "Bucket name must not be null or empty.");
+    Preconditions.checkArgument(key != null && !key.isEmpty(),
+        "COS path must not be null or empty.");
+    Preconditions.checkArgument(client != null, "COSClient must not be null.");
+    mBucketName = bucketName;
+    mKey = key;
+    mCosClient = client;
+
+    mFile = new File(PathUtils.concatPath(CommonUtils.getTmpDir(), UUID.randomUUID()));
+
+    try {
+      mHash = MessageDigest.getInstance("MD5");
+      mLocalOutputStream =
+          new BufferedOutputStream(new DigestOutputStream(new FileOutputStream(mFile), mHash));
+    } catch (NoSuchAlgorithmException e) {
+      LOG.warn("Algorithm not available for MD5 hash.", e);
+      mHash = null;
+      mLocalOutputStream = new BufferedOutputStream(new FileOutputStream(mFile));
+    }
+  }
+
+  /**
+   * Writes the given bytes to this output stream. Before close, the bytes are all written to local
+   * file.
+   *
+   * @param b the bytes to write
+   */
+  @Override
+  public void write(int b) throws IOException {
+    mLocalOutputStream.write(b);
+  }
+
+  /**
+   * Writes the given byte array to this output stream. Before close, the bytes are all written to
+   * local file.
+   *
+   * @param b the byte array
+   */
+  @Override
+  public void write(byte[] b) throws IOException {
+    mLocalOutputStream.write(b, 0, b.length);
+  }
+
+  /**
+   * Writes the given number of bytes from the given byte array starting at the given offset to this
+   * output stream. Before close, the bytes are all written to local file.
+   *
+   * @param b the byte array
+   * @param off the start offset in the data
+   * @param len the number of bytes to write
+   */
+  @Override
+  public void write(byte[] b, int off, int len) throws IOException {
+    mLocalOutputStream.write(b, off, len);
+  }
+
+  /**
+   * Flushes this output stream and forces any buffered output bytes to be written out. Before
+   * close, the data are flushed to local file.
+   */
+  @Override
+  public void flush() throws IOException {
+    mLocalOutputStream.flush();
+  }
+
+  /**
+   * Closes this output stream. When an output stream is closed, the local temporary file is
+   * uploaded to COS Service. Once the file is uploaded, the temporary file is deleted.
+   */
+  @Override
+  public void close() throws IOException {
+    if (mClosed.getAndSet(true)) {
+      return;
+    }
+    mLocalOutputStream.close();
+    try (BufferedInputStream in = new BufferedInputStream(new FileInputStream(mFile))) {
+      ObjectMetadata meta = new ObjectMetadata();
+      meta.setContentLength(mFile.length());
+      if (mHash != null) {
+        byte[] hashBytes = mHash.digest();
+        meta.setContentMD5(new String(Base64.encodeBase64(hashBytes)));
+      }
+      mCosClient.putObject(mBucketName, mKey, in, meta);
+      mFile.delete();
```

Comment:
should `mFile` be outside the `try` block? otherwise, `mFile` is deleted before `in` is closed, not sure if there will be any issue on this

Reply:
If the stream named in is closed, We should delete the local temporary file named mFile.